clients/vscode/src/Commands.ts

```diff
@@ -286,16 +286,21 @@ export class Commands {
             alwaysShow: true,
           });
         }
-        const recentlyCommandToAdd = recentlyCommand.filter((item) => !list.find((i) => i.value === item));
+        const recentlyCommandToAdd = recentlyCommand
+          .filter((item) => !list.find((i) => i.value === item))
+          .slice(0, this.config.numRecentlyCommandsHistory);
```

Comment:
I think we don't need to slice it again, as Line 266 already make this `recentlyCommand` less than this limit.